pennylane/beta/plugins/default_tensor.py

```diff
@@ -337,34 +343,34 @@ def _add_state_prep_nodes(self, operation, par):
 
         self._add_initial_state_nodes(tensors, wires_seq, name)
 
-    def _add_gate_nodes(self, operation, wires, par):
+    def _add_gate_nodes(self, operation, registers, par):
         """Add tensor network nodes and edges related to the quantum gates.
 
         Args:
             operation (str): name of the gate operation
-            wires (Sequence[int]): subsystems the gate is applied on
+            registers (Sequence[int]): indices of wires that the gate is applied on
             par (tuple): parameter values for the gate
         """
         A = self._get_operator_matrix(operation, par)
-        num_wires = len(wires)
+        num_wires = len(registers)
         A = self._reshape(A, [2] * num_wires * 2)
-        op_node = self._add_node(A, wires=wires, name=operation)
+        op_node = self._add_node(A, registers=registers, name=operation)
 
         if self._rep == "exact":
-            for idx, w in enumerate(wires):
+            for idx, w in enumerate(registers):
                 tn.connect(op_node[num_wires + idx], self._free_wire_edges[w])
                 self._free_wire_edges[w] = op_node[idx]
         elif self._rep == "mps":
-            if len(wires) == 1:
-                wire = wires[0]
-                self.mps.apply_one_site_gate(op_node, wire)
-                self._free_wire_edges[wire] = self.mps.nodes[wire][1]
-            elif len(wires) == 2:
-                if abs(wires[1] - wires[0]) == 1:
+            if len(registers) == 1:
+                reg = registers[0]
+                self.mps.apply_one_site_gate(op_node, reg)
+                self._free_wire_edges[reg] = self.mps.nodes[reg][1]
+            elif len(registers) == 2:
+                if abs(registers[1] - registers[0]) == 1:
                     # TODO: set ``max_singular_values`` or ``max_truncation_error``
-                    self.mps.apply_two_site_gate(op_node, *wires)
-                    for wire in wires:
-                        self._free_wire_edges[wire] = self.mps.nodes[wire][1]
+                    self.mps.apply_two_site_gate(op_node, *registers)
+                    for reg in registers:
+                        self._free_wire_edges[reg] = self.mps.nodes[reg][1]
```

Comment:
This is one place where I suspect things would break if `wire_map` did not return a consecutive sequence of indices, starting from zero. The reason for that is because `self.mps.nodes` is an object defined in the `tensornetwork` library, which implicitly indexes the nodes that way

Reply:
I will add tests for this! But since the results of wire_map are indices, I would not see a user-invoked case where this can go  wrong, as long as the device is coded up correctly!